nixos/modules/services/networking/adguardhome.nix

```diff
@@ -12,23 +13,19 @@ let
     "--config /var/lib/AdGuardHome/AdGuardHome.yaml"
   ] ++ cfg.extraArgs);
 
-  configFile = pkgs.writeTextFile {
+  settings = if (cfg.settings != null)
```

Comment:
I was actually quite happy with being able to remove all explicit/external options that influence `cfg.settings`, but I can see that it makes sense to provide a minimal stable interface here.